src/GafferScene/ScenePlug.cpp

```diff
@@ -271,32 +245,49 @@ const PathMatcherDataPlug *ScenePlug::setPlug() const
 	return getChild<PathMatcherDataPlug>( 7 );
 }
 
+ScenePlug::PathScope::PathScope( const Gaffer::Context *context, const ScenePath &scenePath )
+	:	EditableScope( context )
+{
+	set( scenePathContextName, scenePath );
+}
+
+ScenePlug::SetScope::SetScope( const Gaffer::Context *context, const IECore::InternedString &setName )
+	:	EditableScope( context )
+{
+	remove( Filter::inputSceneContextName );
+	remove( ScenePlug::scenePathContextName );
+	set( setNameContextName, setName );
+}
+
+ScenePlug::GlobalScope::GlobalScope( const Gaffer::Context *context )
+	:	EditableScope( context )
+{
+	remove( Filter::inputSceneContextName );
+	remove( scenePathContextName );
+	remove( setNameContextName );
+}
+
 Imath::Box3f ScenePlug::bound( const ScenePath &scenePath ) const
 {
-	ContextPtr tmpContext = new Context( *Context::current(), Context::Borrowed );
-	tmpContext->set( scenePathContextName, scenePath );
-	Context::Scope scopedContext( tmpContext.get() );
+	PathScope scope( Context::current(), scenePath );
 	return boundPlug()->getValue();
 }
 
 Imath::M44f ScenePlug::transform( const ScenePath &scenePath ) const
 {
-	ContextPtr tmpContext = new Context( *Context::current(), Context::Borrowed );
-	tmpContext->set( scenePathContextName, scenePath );
-	Context::Scope scopedContext( tmpContext.get() );
+	PathScope scope( Context::current(), scenePath );
 	return transformPlug()->getValue();
 }
 
 Imath::M44f ScenePlug::fullTransform( const ScenePath &scenePath ) const
 {
-	ContextPtr tmpContext = new Context( *Context::current(), Context::Borrowed );
-	Context::Scope scopedContext( tmpContext.get() );
+	Context::EditableScope tmpContext( Context::current() );
```

Comment:
Even if we don't go with the helper setters I suggested above, maybe it'd be good to use a PathScope here anyways, to clarify the intention? It doesn't much matter for the Path case since there aren't any removes, but if someone used this as example code for a loop involving a `SetScope` they might mistakenly miss those removes.

PS. Why is this variable `tmpContext` instead of `scope`?